crates/runtime/src/catalogconnector/glue.rs

```diff
@@ -14,46 +14,67 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-use super::{CatalogConnector, ParameterSpec, Parameters};
+use super::{CatalogConnector, ParameterSpec};
 use crate::{
     Runtime,
     component::catalog::Catalog,
-    dataconnector::{ConnectorComponent, ConnectorParams},
+    dataconnector::{
+        ConnectorComponent,
+        parameters::{
+            self, ConnectorParams,
+            aws::{AuthValidator, RegionValidator, Validator},
+        },
+        s3,
+    },
 };
 use async_trait::async_trait;
-use aws_config::{BehaviorVersion, Region, SdkConfig};
-use aws_sdk_glue::{
-    Client,
-    error::SdkError,
-    operation::{get_databases::GetDatabasesError, get_tables::GetTablesError},
-    types::Table,
-};
-use aws_sdk_sts::config::{Credentials, ProvideCredentials};
-use data_components::RefreshableCatalogProvider;
-use datafusion::{
-    catalog::{CatalogProvider, SchemaProvider, TableProvider},
-    common::Result as DFResult,
-    error::DataFusionError,
-};
-use globset::GlobSet;
-use iceberg::{
-    NamespaceIdent, TableIdent,
-    io::{FileIOBuilder, S3_ACCESS_KEY_ID, S3_SECRET_ACCESS_KEY},
-    spec::TableMetadata,
-    table::Table as IcebergTable,
-};
-use iceberg_datafusion::IcebergTableProvider;
 use snafu::prelude::*;
-use std::fmt;
-use std::{any::Any, collections::HashMap, sync::Arc};
+use std::any::Any;
+use std::sync::{Arc, LazyLock};
+
+mod provider;
+mod state;
+
+use provider::GlueCatalogProvider;
+
+pub static PREFIX: &str = "glue";
+
+pub static PARAMETERS: LazyLock<Vec<ParameterSpec>> = LazyLock::new(|| {
+    vec![
+        ParameterSpec::component("region")
+            .description("The AWS region for Glue operations")
+            .secret(),
+        ParameterSpec::component("key")
+            .description("The AWS access key ID for Glue authentication")
+            .secret(),
+        ParameterSpec::component("secret")
+            .description("The AWS secret access key for Glue authentication")
+            .secret(),
+        ParameterSpec::component("session_token")
+            .description("The AWS session token for Glue authentication")
+            .secret(),
+    ]
+    .into_iter()
+    .chain(s3::PARAMETERS.iter().cloned())
+    .collect()
+});
+
+static VALIDATORS: LazyLock<Vec<Box<dyn Validator + Send + Sync + 'static>>> =
+    LazyLock::new(|| vec![Box::new(RegionValidator), Box::new(AuthValidator)]);
 
 #[derive(Debug, Snafu)]
 pub enum Error {
     #[snafu(display("Failed to get Glue databases: {source}"))]
-    GetDatabases { source: SdkError<GetDatabasesError> },
```

Comment:
Is there a reason we're using full imports (e.g. `aws_sdk_glue::error::SdkError`) over importing at top of file?